src/pages/blog/[slug].tsx

```diff
@@ -0,0 +1,46 @@
+import * as Sentry from '@sentry/nextjs'
+import { GetStaticPaths, GetStaticProps } from 'next'
+import BlogPostPage from 'components/blog/BlogPostPage'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import { createGhostClient } from 'common/util/ghost-client'
+
+export const getStaticProps: GetStaticProps = async ({ params, locale }) => {
+  if (typeof params?.slug !== 'string') return { notFound: true }
+
+  try {
+    const client = createGhostClient()
+    const post = await client.posts.read({ slug: params.slug })
+
+    if (!post) {
+      return { notFound: true }
+    }
+    return {
+      props: {
+        post,
+        ...(await serverSideTranslations(locale ?? 'bg', ['common', 'blog'])),
+      },
+    }
+  } catch (error) {
+    console.error(error)
+    Sentry.captureException(error)
+    return { notFound: true }
+  }
+}
```

Comment:
If we want to have good SEO we need to have one of the following two approaches:

- SSR hydratation via `getStaticProps` and `getStaticPaths`
- SSR hydratation via `getServerSideProps`

In order to have the page content downloaded from the  server and embedded in the HTML we need to practice the methods above.

For the blog SP is more appropriate as content changes rarely.
For the campaigns SSP seems to be more appropriate since content needs to be up-to-date often.